File: app/src/main/java/org/code4projects/framework/impl/AndroidPixmap.java
/*
 * Copyright (c) 2016-2026 Salvatore D'Angelo
 * Licensed under the MIT License. See the LICENSE file in the project root.
 */
package org.code4projects.framework.impl;

import android.graphics.Bitmap;

import org.code4projects.framework.Graphics.PixmapFormat;
import org.code4projects.framework.Pixmap;

/*
 * This class represents a bitmap. On Android a bitmap is managed by a Bitmap class.
 * Each bitmap contains also the format on how its colors are stored (PixmapFormat).
 *
 * @author mzechner
 */
public class AndroidPixmap implements Pixmap {
    Bitmap bitmap;
    PixmapFormat format;

    /*
     * Initializes the bitmap.
     */
    public AndroidPixmap(Bitmap bitmap, PixmapFormat format) {
        this.bitmap = bitmap;
        this.format = format;
    }

    /*
     * Gets the width of the bitmap.
     */
    public int getWidth() {
        return bitmap.getWidth();
    }

    /*
     * Gets the height of the bitmap.
     */
    public int getHeight() {
        return bitmap.getHeight();
    }

    /*
     * Gets the format of the bitmap. Possible values are: ARGB8888, ARGB4444, RGB565.
     */
    public PixmapFormat getFormat() {
        return format;
    }

    /*
     * Disposes a bitmap.
     */
    public void dispose() {
        bitmap.recycle();
    }

    /*
     * Gets the color of the pixel in (x, y) position.
     */
    public int getColor(int x, int y) {
        return bitmap.getPixel(x, y);
    }

    /*
     * Sets the color of the pixel in (x, y) position.
     */
    public void setColor(int x, int y, int color) {
        bitmap.setPixel(x, y, color);
    }
}

File: app/src/main/kotlin/org/code4projects/framework/impl/AndroidPixmap.kt
/*
 * Copyright (c) 2016-2026 Salvatore D'Angelo
 * Licensed under the MIT License. See the LICENSE file in the project root.
 */
package org.code4projects.framework.impl

import android.graphics.Bitmap

import org.code4projects.framework.Graphics
import org.code4projects.framework.Pixmap

/*
 * This class represents a bitmap. On Android a bitmap is managed by a Bitmap class.
 * Each bitmap contains also the format on how its colors are stored (PixmapFormat).
 *
 * @author mzechner
 */
class AndroidPixmap(val bitmap: Bitmap, private val format: Graphics.PixmapFormat) : Pixmap {

    /*
     * Gets the width of the bitmap.
     */
    override fun getWidth(): Int = bitmap.width

    /*
     * Gets the height of the bitmap.
     */
    override fun getHeight(): Int = bitmap.height

    /*
     * Gets the format of the bitmap. Possible values are: ARGB8888, ARGB4444, RGB565.
     */
    override fun getFormat(): Graphics.PixmapFormat = format

    /*
     * Disposes a bitmap.
     */
    override fun dispose() {
        bitmap.recycle()
    }

    /*
     * Gets the color of the pixel in (x, y) position.
     */
    override fun getColor(x: Int, y: Int): Int = bitmap.getPixel(x, y)

    /*
     * Sets the color of the pixel in (x, y) position.
     */
    override fun setColor(x: Int, y: Int, color: Int) {
        bitmap.setPixel(x, y, color)
    }
}
